gu-net/src/rpc/peer.rs

```diff
@@ -130,3 +177,91 @@ impl Handler<GetPeer> for PeerManager {
         MessageResult(self.peers.get(&msg.0).cloned())
     }
 }
+
+// Tag management
+pub struct AddTags {
+    pub node: NodeId,
+    pub tags: Tags,
+}
+
+impl Message for AddTags {
+    type Result = Option<()>;
+}
+
+impl Handler<AddTags> for PeerManager {
+    type Result = Option<()>;
```

Comment:
Maybe Result<()>?

Reply:
I changed it to `Result<(), ()>` but this is not really expressive, but I'm not sure if I should add a new type just to convey "Peer not found". On the other hand, `Result<(), String>` doesn't seem clean.